### DISCUSSION TITLE
Panda CSS + ReScript

### DISCUSSION THREAD

**Discussion by vmarcosp** [chakra-ui/panda#1535] [Q&A] [answered]
Hey everyone!
I'm trying to create ReScript bindings (similar to type definitions in TypeScript) for Panda but I'm facing a issue with the generated code (generated by the ReScript compiler). I'll not share the bindings because this is not relevant for the question, but the generated code is something like that:
```javascript
import * as React from "react";
import * as PandaCSS from "./styled-system/css";
import * as Client from "react-dom/client";

import './index.css';

function App(props) {
    return React.createElement("div", {
        className: PandaCSS.css({
            display: "block"
        })
    }, "Hello");
}

````

Checking the devtools and HTML, I can see the element with the proper class applied, but the element doesn't have the `display:block` styles as you can see here:

<img width="1403" alt="Screen Shot 2023-10-17 at 08 16 53" src="https://github.com/chakra-ui/panda/assets/20327229/c3524312-4717-4a13-83f7-1a197e7e486e">

Changing the PandaCSS import from `import * as PandaCSS from "./styled-system/css"` to `import { css } from "./styled-system/css" eveything works as expected. My question here is: The css functions (and other utlities) should be always imported as named imports? That's why the original generated code isn't working? 
Thanks in advance.

**Comment by astahmer** [comment, marked as answer]
you're right, currently the static extractor does not handle namespace imports !

is that a blocker for you ? I could look into implementing it if needed

**Reply by vmarcosp** [reply]
Thanks for answering @astahmer.
Yep, it's a blocker, but it's not that urgent. Additionally, I'd like to contribute to the project by adding support for namespace imports. I can open an issue and try to start with a small PR.

**Reply by astahmer** [reply]
awesome ! the related code is around here https://github.com/chakra-ui/panda/blob/8a3bf6d9677505d825ef79656650cd2419b96412/packages/parser/src/import.ts#L29

**Reply by vmarcosp** [reply]
Great. I was looking at extractor/extract.ts to see where the css(...) function call get its arguments extracted to static CSS: https://github.com/chakra-ui/panda/blob/main/packages/extractor/src/extract.ts#L162.
Debugging using the same code I provided in the examples, I realized that the `fnName` doesn't match the `css` function name as expected. Should I add support for namespace imports in this part as well?

**Reply by astahmer** [reply]
hm I'm not sure why the extractor needs to be touched but go ahead if you feel like it's needed as well !

**Reply by segunadebayo** [reply]
Namespace imports are now supported in Panda 🎉
